src/main/java/bio/terra/service/dataset/DatasetRequestValidator.java

```diff
@@ -363,6 +367,30 @@ private void validateSchema(DatasetSpecificationModel schema, Errors errors) {
         }
     }
 
+    private void validateRegion(DatasetRequestModel datasetRequest, Errors errors) {
+        if (datasetRequest.getRegion() != null) {
+            if (datasetRequest.getCloudPlatform() == null) {
+                errors.rejectValue("region", "InvalidRegionForPlatform",
+                    "Cannot set a region when a cloudPlatform is not provided.");
+            }
+            boolean supported = false;
+            switch (datasetRequest.getCloudPlatform()) {
+                case GCP:
+                    supported = SUPPORTED_GOOGLE_REGIONS.contains(datasetRequest.getRegion().toLowerCase());
+                    break;
+                case AZURE:
+                    errors.rejectValue("cloudPlatform", "InvalidCloudPlatform",
```

Comment:
Probably want a `return` here otherwise you’ll get another error about valid regions which has no meaning for the azure platform.